File: 386. Lexicographical Numbers/solution.py
class Solution:
    def dfs(self, curr: int, n: int, ans: List[int]):
        if curr > n:
            return

        ans.append(curr)

        for i in range(10):
            if curr * 10 + i > n:
                return
            self.dfs(curr * 10 + i, n, ans)

    def lexicalOrder(self, n: int) -> List[int]:
        ans = []
        for i in range(1, 10):
            self.dfs(i, n, ans)
        return ans

File: 386. Lexicographical Numbers/solution.cpp
class Solution {
public:
    void dfs(int curr,int n, vector<int>&ans){

        if(curr>n)
            return;
        
        ans.push_back(curr);
        
        for(int i=0;i<=9;i++){

            if(curr*10 + i > n)
                return;

            dfs(curr*10+i,n,ans);
        }
    }

    vector<int> lexicalOrder(int n) {
        vector<int>ans;
        for(int i=1;i<10;i++)
            dfs(i,n,ans);  

        return ans;
    }
};